DisguiseUnityRenderStream/Runtime/ScratchTextureManager.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using UnityEngine;
+using Debug = UnityEngine.Debug;
+
+namespace Disguise.RenderStream
+{
+    /// <summary>
+    /// A simpler version of <see cref="Disguise.RenderStream.TemporaryTexture2DManager"/> that doesn't need to manage texture lifetime.
+    /// </summary>
+    public abstract class ScratchTextureManager<TTexture>
+    {
+        readonly Dictionary<Texture2DDescriptor, TTexture> m_Items = new Dictionary<Texture2DDescriptor, TTexture>();
+
+        protected string m_Name;
+        
+        public void Clear()
+        {
+            DebugLog($"Cleared textures");
+
+            foreach (var texture in m_Items.Values)
+            {
+                DestroyTexture(texture);
+            }
+            
+            m_Items.Clear();
+        }
+        
+        public TTexture Get(Texture2DDescriptor descriptor)
+        {
+            if (m_Items.TryGetValue(descriptor, out var item))
+            {
+                return item;
+            }
+
+            DebugLog($"Created texture: {descriptor}");
+            
+            var texture = CreateTexture(descriptor);
+            m_Items.Add(descriptor, texture);
+            return texture;
+        }
+
+        protected abstract TTexture CreateTexture(Texture2DDescriptor descriptor);
+        
+        protected abstract void DestroyTexture(TTexture texture);
+        
+        [Conditional("DISGUISE_LOG_TEXTURE_POOLING")]
```

Comment:
Suggestion: `DISGUISE_VERBOSE_LOGGING`. This is definitely something we can use elsewhere in the plugin in the future.